elements/lisk-api-client/src/api_client.ts

```diff
@@ -12,163 +12,44 @@
-import { VotesResource } from './resources/votes';
-
-const defaultOptions = {
-	bannedNodes: [],
-	randomizeNodes: true,
-};
-
-const commonHeaders: HashMap = {
-	Accept: 'application/json',
-	'Content-Type': 'application/json',
-};
-
-const getClientHeaders = (clientOptions: ClientOptions): HashMap => {
-	const { name = '????', version = '????', engine = '????' } = clientOptions;
-
-	const liskElementsInformation = 'LiskElements/1.0 (+https://github.com/LiskHQ/lisk-elements)';
-	const locale: string | undefined =
-		process.env.LC_ALL ?? process.env.LC_MESSAGES ?? process.env.LANG ?? process.env.LANGUAGE;
-	const systemInformation = `${os.platform()} ${os.release()}; ${os.arch()}${
-		locale ? `; ${locale}` : ''
-	}`;
-
-	return {
-		'User-Agent': `${name}/${version} (${engine}) ${liskElementsInformation} ${systemInformation}`,
-	};
-};
-
-export interface ClientOptions {
-	readonly engine?: string;
-	readonly name?: string;
-	readonly version?: string;
-}
+import { EventCallback, Channel, RegisteredSchema, NodeInfo } from './types';
+import { Node } from './node';
 
 export class APIClient {
-	public accounts: AccountsResource;
-	public bannedNodes!: ReadonlyArray<string>;
-	public blocks: BlocksResource;
-	public currentNode!: string;
-	public dapps: DappsResource;
-	public delegates: DelegatesResource;
-	public headers!: HashMap;
-	public node: NodeResource;
-	public nodes!: ReadonlyArray<string>;
-	public peers: PeersResource;
-	public randomizeNodes!: boolean;
-	public transactions: TransactionsResource;
-	public voters: VotersResource;
-	public votes: VotesResource;
-
-	public constructor(nodes: ReadonlyArray<string>, providedOptions: InitOptions = {}) {
-		this.initialize(nodes, providedOptions);
-		this.accounts = new AccountsResource(this);
-		this.blocks = new BlocksResource(this);
-		this.dapps = new DappsResource(this);
-		this.delegates = new DelegatesResource(this);
-		this.node = new NodeResource(this);
-		this.peers = new PeersResource(this);
-		this.transactions = new TransactionsResource(this);
-		this.voters = new VotersResource(this);
-		this.votes = new VotesResource(this);
-	}
-
-	public static get constants(): typeof constants {
-		return constants;
-	}
-
-	public static createMainnetAPIClient(options?: InitOptions): APIClient {
-		return new APIClient(constants.MAINNET_NODES, {
-			genesisBlockTransactionRoot: constants.MAINNET_NETHASH,
-			...options,
-		});
-	}
+	private readonly _channel: Channel;
+	private _schemas!: RegisteredSchema;
+	private _nodeInfo!: NodeInfo;
+	private _node!: Node;
 
-	public static createTestnetAPIClient(options?: InitOptions): APIClient {
-		return new APIClient(constants.TESTNET_NODES, {
-			genesisBlockTransactionRoot: constants.TESTNET_NETHASH,
-			...options,
-		});
+	public constructor(channel: Channel) {
+		this._channel = channel;
 	}
 
-	public banActiveNode(): boolean {
-		return this.banNode(this.currentNode);
+	public async connect(): Promise<void> {
```

Comment:
Will this method carry over extra tasks once more components of the client are implemented? 
Asking right now it's only invoking some actions and instantiating nodes and the name `connect()` usually implies opening some sort of communication channel.

Reply:
It was calling `channel.connect` in the draft, but actually if we accept opened channel as input for `createClient` it cannot call `connect` internally anymore.
I will update to `init`